### ISSUE TITLE
center loss doesn't seem to decay

### ISSUE THREAD

**Issue by mxmxlwlw** [davidsandberg/facenet#575]
Hi, I've tried the center loss but it seems it is always equal to 0.001953125. Although, it do has the effect to the loss.
Here's the code:
```
def center_loss(features, label, alfa, nrof_classes):
    with tf.variable_scope("center_loss"):
        nrof_features = features.get_shape()[1]
        centers = tf.get_variable('centers', [nrof_classes, nrof_features], dtype=tf.float32,
            initializer=tf.constant_initializer(0), trainable=False)
        label = tf.reshape(label, [-1])
        centers_batch = tf.gather(centers, label)
        diff = (1 - alfa) * (centers_batch - features)
        centers = tf.scatter_sub(centers, label, diff)
        loss = tf.reduce_mean(tf.square(features - centers_batch))
        return loss, centers
embeddings = tf.nn.l2_normalize(end_points['feat'], 1, 1e-10, name='embeddings')
center_loss, _ = center_loss(embeddings, labels, 0.95, classes)
loss = tf.add_n([l2loss, center_loss, cross_entropy_mean])
```

**Comment by Vallum** [comment]
I think it's a bug. There is no "centers"  update source node in the defined graph.
In your example as well, the centers return to "_" (null), so the centers begins from the default value at each batch round.
In this case, the centers doesn't update at all, so you need to add control dependency to update centers.
